src/lib/components/Widget.tsx

```diff
@@ -111,15 +111,16 @@ export default function Widget(props: PropsWithChildren<WidgetProps>) {
           <WidgetWrapper width={width} className={className} ref={wrapper}>
             <DialogProvider value={dialog || wrapper.current}>
               <ErrorBoundary onError={onError}>
-                <ReduxProvider store={multicallStore}>
-                  <AtomProvider>
-                    <Web3Provider provider={provider} jsonRpcEndpoint={jsonRpcEndpoint}>
-                      <Updaters />
-                      <ErrorGenerator {...props} />
-                      {children}
-                    </Web3Provider>
-                  </AtomProvider>
-                </ReduxProvider>
+                <WidgetPropValidator {...props}>
```

Comment:
Can this be made a sibling DOM node, like the `<Updaters />`, to reduce nesting?

Reply:
I don't think so, since the child tree is intentionally not rendered until props have been validated.